src/coreclr/jit/morph.cpp

```diff
@@ -9675,7 +9675,25 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac, bool* optA
                 // We do not need to throw if the second operand is a non-(negative one) constant.
                 if (!op2->IsIntegralConst() || op2->IsIntegralConst(-1))
                 {
-                    fgAddCodeRef(compCurBB, bbThrowIndex(compCurBB), SCK_OVERFLOW);
+                    bool checkDividend = true;
+
+                    // We do not need to throw if we know the first operand is a constant and not a min-value.
+                    if (op1->IsIntegralConst())
+                    {
+                        if (tree->TypeIs(TYP_INT) && !op1->IsIntegralConst(INT32_MIN))
+                        {
+                            checkDividend = false;
+                        }
+                        else if (tree->TypeIs(TYP_LONG) && !op1->IsIntegralConst(INT64_MIN))
+                        {
+                            checkDividend = false;
+                        }
+                    }
+
+                    if (checkDividend)
```

Comment:
NOTE: this still leaves overflow block alive if it's unused but presumably it's a different issue

Reply:
Hhm, won't it be dead code eliminated?

Reply:
Throw helper blocks are marked as no-remove. This is because they aren't rooted in the normal flowgraph (the edges from the throwing node to the throw helper block are implicit).

Reply:
> Hhm, won't it be dead code eliminated?

these blocks are special, they have BBF_DONT_REMOVE or something like that. We spawn them early in importer/morph and then tie to corresponding divisions if needed, it's too late to remove them in codegen

Reply:
I'm hopeful that this PR will resolve those cases as it's already happening today.